modules/tinymce/src/core/test/ts/webdriver/paste/CopyAndPaste.ts

```diff
@@ -0,0 +1,143 @@
+  it('TINY-7719: Wrapped elements are preserved in copy and paste (pre + headings)', async () => {
+    const editor = hook.editor();
+
+    const testBlockTags = async (tagName) => {
+      editor.setContent(
+        `<${tagName}>abc</${tagName}>` +
+        '<p>other kind of tag</p>' +
+        `<${tagName}>same tag</${tagName}>`
+      );
+      await copyAndPast(
+        editor,
+        [[ 0, 0 ], 0, [ 0, 0 ], 3 ],
+        [[ 2, 0 ], 1, [ 2, 0 ], 3 ]
+      );
+      await copyAndPast(
+        editor,
+        [[ 0, 0 ], 0, [ 0, 0 ], 3 ],
+        [[ 1, 0 ], 1, [ 1, 0 ], 4 ]
+      );
+      TinyAssertions.assertContent(editor,
+        `<${tagName}>abc</${tagName}>\n` +
+        '<p>o</p>\n' +
+        `<${tagName}>abc</${tagName}>\n` +
+        '<p>r kind of tag</p>\n' +
+        `<${tagName}>s</${tagName}>\n` +
+        `<${tagName}>abc</${tagName}>\n` +
+        `<${tagName}>e tag</${tagName}>`
+      );
+    };
+
+    for (const tagName of [ 'h1', 'h2', 'h3', 'h4', 'h5', 'h6' ]) {
+      await testBlockTags(tagName);
+    }
+
+    const testPreTag = async (tagName) => {
+      editor.setContent(
+        `<${tagName}>abc</${tagName}>` +
+        '<p>other kind of tag</p>' +
+        `<${tagName}>same tag</${tagName}>`
+      );
+      await copyAndPast(
+        editor,
+        [[ 0, 0 ], 0, [ 0, 0 ], 3 ],
+        [[ 2, 0 ], 1, [ 2, 0 ], 3 ]
+      );
+      await copyAndPast(
+        editor,
+        [[ 0, 0 ], 0, [ 0, 0 ], 3 ],
+        [[ 1, 0 ], 1, [ 1, 0 ], 4 ]
+      );
+      TinyAssertions.assertContent(editor,
+        `<${tagName}>abc</${tagName}>\n` +
+        '<p>o</p>\n' +
+        `<${tagName}>abc</${tagName}>\n` +
+        '<p>r kind of tag</p>\n' +
+        `<${tagName}>sabce tag</${tagName}>`
+      );
+    };
+
+    for (const tagName of [ 'pre' ]) {
+      await testPreTag(tagName);
+    }
+  });
+
+  it('TINY-7719: Wrapped elements are preserved in copy and paste (inline elements)', async () => {
+    const editor = hook.editor();
+
+    const testInlineTags = async (tagName) => {
+      editor.setContent(
+        `<p><${tagName} class="someclass">abc</${tagName}></p>` +
+        '<h1>something</h1>' +
+        '<p>abc def</p>'
+      );
+      await copyAndPast(
+        editor,
+        [[ 0, 0, 0 ], 0, [ 0, 0, 0 ], 3 ],
+        [[ 1, 0 ], 1, [ 1, 0 ], 4 ]
+      );
+      TinyAssertions.assertContent(editor,
+        `<p><${tagName} class="someclass">abc</${tagName}></p>\n` +
+        `<h1>s<${tagName} class="someclass">abc</${tagName}>thing</h1>\n` +
+        '<p>abc def</p>'
+      );
+    };
+
+    const inlineElements = [
+      'big',
+      'small',
+      'tt',
+      'abbr',
+      'acronym',
+      'cite',
+      'code',
+      'dfn',
+      'em',
+      'kbd',
+      'strong',
+      'samp',
+      'var',
+      'a',
+      'bdo',
+      'map',
+      'q',
+      'span',
+      'sub',
+      'sup',
+      'button',
+      'label'
+    ];
```

Comment:
This is quite verbose and takes up a lot of room... it could be squashed down to something like this:
```suggestion
    const inlineElements = [
      'big', 'small', 'tt', 'abbr', 'acronym', 'cite', 'code',
      'dfn', 'em', 'kbd', 'strong', 'samp', 'var', 'a', 'bdo',
      'map', 'q', 'span', 'sub', 'sup', 'button', 'label'
    ];
```

Reply:
done